app/models/user.rb

```diff
@@ -15,7 +15,7 @@ class User < ApplicationRecord
 
   def friends
```

Comment:
It is better to use scopes to handle friends.
You can have confirmed friendship relations where confirmed is true and have friends relation through that confirmed friendship.

It is less complex this way and makes proper use of your reverse row addition in `accept_friend`.

You can also get pending friendships in same manner with the foreign key of `user_id` and were confirmed is false ( friendships that you sent. You will again define pending friends through the pending_friendships relation.
Example:
```ruby
has_many :pending_friendships, -> { where confirmed: false }, class_name: "Friendship", foreign_key: "user_id"
has_many :pending_friends, through: :pending_friendships, source: :inviter
```

and inverted friendships (requests you seent (line 26) in same manner.

These additional methods would not be necessary and your code would be much more efficient.